ersilia_precalc_poc/models.py

```diff
@@ -1,11 +1,17 @@
-from dataclasses import dataclass
+import pandera as pa
+from pydantic import BaseModel
 
 
-@dataclass
-class Prediction:
+class Prediction(BaseModel):
     """Dataclass to represent a single prediction"""
 
     model_id: str
     input_key: str
     smiles: str
     output: float
+
+
+class PredictionSchema(pa.DataFrameModel):
```

Comment:
pretty cool how this works -- define a schema here as a class, which we then use as our type for any functions we expect to pass data frames into...